api/src/paths/dwc/validate.ts

```diff
@@ -161,26 +77,78 @@ POST.apiDoc = {
   }
 };
 
-function prepDWCArchive(): RequestHandler {
+function getSubmissionS3Key(): RequestHandler {
   return async (req, res, next) => {
-    defaultLog.debug({ label: 'prepDWCArchive', message: 'files.length', files: req?.files?.length });
+    defaultLog.debug({ label: 'getSubmissionS3Key', message: 'params', files: req.body });
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    const occurrenceSubmissionId = req.body.occurrence_submission_id;
 
-    if (!req.files || !req.files.length) {
-      // no media objects included
-      throw new HTTP400('Missing upload data');
+    if (!occurrenceSubmissionId) {
+      throw new HTTP400('Missing required body param `occurrence_submission_id`.');
     }
 
-    if (req.files.length !== 1) {
-      // no media objects included
-      throw new HTTP400('Too many files uploaded, expected 1.');
+    try {
+      const sqlStatement = getSurveyOccurrenceSubmissionSQL(occurrenceSubmissionId);
+
+      if (!sqlStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
+
+      const response = await connection.query(sqlStatement.text, sqlStatement.values);
+
+      if (!response || !response.rowCount) {
```

Comment:
we should check for existence of `response.rows` and maybe `response.rows[0]` here so we can confidently get the key from it below

Reply:
I double checked what rowCount is, and my understanding had been a bit off.  rowCount is the number of affected rows, but not necessarily the number of rows returned, which makes sense.  So ya, changing this to check for rows is appropriate.